src/ipc/ipc4/handler.c

```diff
@@ -89,6 +89,8 @@ static inline void ipc4_send_reply(struct ipc4_message_reply *reply)
 {
 	struct ipc *ipc = ipc_get();
 
+	/* copy the extension from the message reply */
+	reply->extension.dat = msg_reply.extension;
```

Comment:
this change will affect the "normal" on-DSP use-case?

Reply:
@lyakh This is under ifdef CONFIG_LIBRARY....